net-analyzer/goaccess/goaccess-1.6.4.ebuild

```diff
@@ -0,0 +1,70 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit autotools optfeature systemd tmpfiles
+
+DESCRIPTION="A real-time web log analyzer and interactive viewer in a terminal"
+HOMEPAGE="https://goaccess.io"
+SRC_URI="https://tar.goaccess.io/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="debug geoip geoipv2 getline ssl unicode"
+REQUIRED_USE="geoipv2? ( geoip )"
+
+RDEPEND="acct-group/goaccess
+	acct-user/goaccess
+	sys-libs/ncurses:=[unicode(+)?]
+	geoip? (
+		!geoipv2? ( dev-libs/geoip )
+		geoipv2? ( dev-libs/libmaxminddb:0= )
+	)
+	ssl? ( dev-libs/openssl:0= )"
+DEPEND="${RDEPEND}"
+BDEPEND="virtual/pkgconfig"
+
+src_prepare() {
+	default
+
+	# Enable log-format, define log file and db path,
+	# change path to GeoIP bases in config
+	sed -i  -e '/log-format COMBINED/s/#//' \
+		-e '/log-file/s/#//' \
+		-e '/db-path/s|tmp|var/lib/goaccess|' \
+		-e '/geoip-database/s|local/||' config/goaccess.conf \
+		|| die "sed failed for goaccess.conf"
+
+	eautoreconf
+}
+
+src_configure() {
+	local myeconfargs=()
+	readarray -t myeconfargs < <(
+		use_enable debug
+		use_enable geoip geoip "$(usex geoipv2 mmdb legacy)"
+		use_enable unicode utf8
+		use_with getline
+		use_with ssl openssl
+	)
+	econf "${myeconfargs[@]}"
```

Comment:
What is going on here? What's wrong with `$(use_enable)` and `$(use_with)`?

Reply:
They do not pass shellsheck validation in this case.

In case 

```        myeconfargs=(
                $(use_enable debug)
                $(use_enable geoip geoip "$(usex geoipv2 mmdb legacy)")
                $(use_enable unicode utf8)
                $(use_with getline)
                $(use_with ssl openssl)
        )
```
it will be complaining for "SC2207: Prefer mapfile or read -a to split command output (or quote to avoid splitting)."

In case 

```        myeconfargs=(
                '$(use_enable debug)'
                '$(use_enable geoip geoip "$(usex geoipv2 mmdb legacy)")'
                '$(use_enable unicode utf8)'
                '$(use_with getline)'
                '$(use_with ssl openssl)'
        )
```

it will be complaining for "SC2016: Expressions don't expand in single quotes, use double quotes for that."

And double quotes are not suitable with geoip use flag.

readarray does the job.

Reply:
Yeah ebuilds aren't exactly pure shell, and there's a lot of external sourcing going on that I doubt will ever be accounted with shellcheck. Truthfully I haven't heard anyone using shellcheck to check ebuilds. 

But the problem would be tree-wide and should be fixed elsewhere.

Reply:
> Truthfully I haven't heard anyone using shellcheck to check ebuilds.

People uses it.
You can easily proof it yourself with `grep -R "shellcheck disable" /var/db/repos/gentoo/*`

> Yeah ebuilds aren't exactly pure shell, and there's a lot of external sourcing going on that I doubt will ever be accounted with shellcheck.

I know that, but even with keeping this in mind shellcheck helps in finding issues.

> But the problem would be tree-wide and should be fixed elsewhere.

Agree, but here we have to deal with a single ebuild so far.
I have an overlay with 700 ebuilds in it with shellcheck validation against each, including net-analyzer/goaccess.
And I would like continue to keep ebuilds in sync between overlay and portage.
So, it readarray is not acceptable, may I add the following string to ebuild:

`shellcheck disable=SC2207`

?

Reply:
> $ grep -R "shellcheck disable" . | grep ".ebuild" | wc -l
> 5

5 ebuilds, 2 packages, so it's really not that wide-spread. But yes, that seems much better than the current method.

Reply:
Ping.

Reply:
Discussed change has been made in the last commit.